src/interactive/repl.ts

```diff
@@ -376,6 +376,63 @@ function clearProgress() {
     }
 }
 
+function display(params: { kind: string, data: any }) {
+    if (params.kind === 'application/vnd.julia-vscode.diagnostics') {
+        displayDiagnostics(params)
+    } else {
+        plots.displayPlot(params)
+    }
+}
+
+interface diagnosticData {
+    msg: string,
+    path: string,
+    line: number,
+    severity: number,
+    relatedInformation: {msg: string, path: string, line: number}[]
+}
+const g_trace_diagnostics: Map<string, vscode.DiagnosticCollection> = new Map()
+function displayDiagnostics(params: { kind: string, data: { source: string, items: diagnosticData[] } }) {
+    const source = params.data.source
+
+    if (g_trace_diagnostics.has(source)) {
+        g_trace_diagnostics.get(source).clear()
+    } else {
+        g_trace_diagnostics.set(source, vscode.languages.createDiagnosticCollection('Julia Diagnostics: ' + source))
+    }
+
+    const diagnostics = params.data.items.map((frame): [vscode.Uri, vscode.Diagnostic[]] => {
+        const diagnostic = new vscode.Diagnostic(
+            new vscode.Range(frame.line - 1, 0, frame.line - 1, 99999),
```

Comment:
Maybe the start column should just be the first char on that line that isn't a space or tab? I think all our other diagnostics looks "good" in that sense now?

Reply:
Yeah, fair.

Reply:
I've added an additional `range` argument; doing this automatically when only `line` is provided is hard(ish) and brittle.